### ISSUE TITLE
Prevent calling composables directly in the event handler

### ISSUE THREAD

**Issue by Haberkamp** [vuejs/eslint-plugin-vue#2721]
**Please describe what the rule should do:**

This rule enforces that you can't directly call a composable inside an event handler.

Calling a composable inside an event handler directly can cause issues. For example when we call
`onMounted` inside that composable, we'll get an error because `onMounted` is called outside the Vue context.

Here's an [example](https://play.vuejs.org/#eNp9UctOwzAQ/JXFl6QSSg9wKmnFQ0WAxEOAxMWXkGxCWse2/CiVqvw7a0d9SagXy56Z9c7sbtiN1tnKI5uw3Jam1Q4sOq9nXLadVsbBBpR8Vl46rKCH2qgOEipIrg4UBusjjksuay9L1yoJ3uK9UukINlzC/rOUkOlsAAFKJa0SmAnVpEk3KJIR9QDoR+E8EhS2qiPbh06BctDZBqbBSZo8oBAKvpQR1VlC1fl4iEah6OGw06JwSC+A/Ns7RyavS9GWyylng1vOZhGADvPxICF5Pj6oZefMWWpdt022sErSBGMWzkrV6VagedUhv+Vssk3JWUHOfp8i5ozH8y1e/mC5/Adf2HXAOHszaNGskLMd5wrToBvo+ccLrum+IztVeUHqE+Q70kR98DjIbr2syPaBLrp9jFtuZfNp52uH0m5DBaNxQVHPGW3+7kT0vd2L7DLW0f5Y/wfvT9cA)

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
  <button @click="useFoo">Click me!</button>
</template>

<script setup>
function useFoo() {
  onMounted() {}

  fetch('/api);
}
</script>
```

**Additional context**

To determine wether something is a composable, we check if a function starts with `use`. This is the standard convention for Vue.

Note: I would help to create a PR for that rule
